src/NUnitTestAdapter/TestConverter.cs

```diff
@@ -173,51 +217,54 @@ public DiaNavigationData GetNavigationData(string className, string methodName)
         }
 
         // Public for testing
-        public static TestOutcome ResultStateToTestOutcome(ResultState resultState)
+        public static TestOutcome GetTestOutcome(XmlNode resultNode)
         {
-            switch (resultState)
+            switch (resultNode.GetAttribute("result"))
             {
-                case ResultState.Cancelled:
-                    return TestOutcome.None;
-                case ResultState.Error:
-                    return TestOutcome.Failed;
-                case ResultState.Failure:
+                case "Passed":
+                    return TestOutcome.Passed;
+                case "Failed":
                     return TestOutcome.Failed;
-                case ResultState.Ignored:
+                case "Skipped":
                     return TestOutcome.Skipped;
-                case ResultState.Inconclusive:
+                default:
                     return TestOutcome.None;
-                case ResultState.NotRunnable:
-                    return TestOutcome.Failed;
-                case ResultState.Skipped:
-                    return TestOutcome.Skipped;
-                case ResultState.Success:
-                    return TestOutcome.Passed;
             }
-
-            return TestOutcome.None;
         }
 
-        private string GetErrorMessage(NUnitTestResult result)
+        private static readonly string NL = Environment.NewLine;
+
+        private string GetErrorMessage(XmlNode resultNode)
         {
-            string message = result.Message;
-            string NL = Environment.NewLine;
+            XmlNode messageNode = resultNode.SelectSingleNode("failure/message");
+            if (messageNode != null)
+            {
+                string message = messageNode.InnerText;
+
+                // If we're running in the IDE, remove any caret line from the message
+                // since it will be displayed using a variable font and won't make sense.
+                if (!string.IsNullOrEmpty(message) && RunningUnderIDE)
+                {
+                    string pattern = NL + "  -*\\^" + NL;
+                    message = Regex.Replace(message, pattern, NL, RegexOptions.Multiline);
```

Comment:
Possibly a minor point, but if this happens often, we should promote the Regex to a a member variable. At the very least, there is no need to do the string concatenation every time, that could replace the NL const.